### PR TITLE
pallet-membership weights

### PR DESCRIPTION
**Pull request by Dinonard**
## Summary
* use benchamarked weights instead of hardcoded ones for `pallet-membership`
* rename benchmark to match extrinsic name
* remove unnecessary dependency from `clear_prime`

### CODE REVIEW

File: substrate/frame/membership/src/lib.rs [hunk 1]
```diff
@@ -500,8 +516,7 @@ mod benchmark {
 		}
 
 		clear_prime {
-			let m in 1 .. T::MaxMembers::get();
```
Comment (ggwpez) [inline substrate/frame/membership/src/lib.rs:503]: I would not change this without reason since it will cause a breaking change in the `WeightInfo` trait and cause runtimes with it to not compile anymore.

File: substrate/frame/membership/src/migrations/v4.rs [hunk 2]
```diff
@@ -47,7 +47,7 @@ pub fn migrate<T: frame_system::Config, P: GetStorageVersion + PalletInfoAccess,
 			target: LOG_TARGET,
 			"New pallet name is equal to the old prefix. No migration needs to be done.",
 		);
-		return Weight::zero()
+		return Weight::zero();
```
Comment (ggwpez) [inline substrate/frame/membership/src/migrations/v4.rs:50]: Please revert unrelated changes.

**Review by ggwpez — COMMENTED** [review]
Good catch 😅

File: substrate/frame/membership/src/migrations/v4.rs (hunk 2 above)
Comment (Dinonard) [reply substrate/frame/membership/src/migrations/v4.rs:50]: Oh, my bad, this was done by `cargo fmt` 😅

File: substrate/frame/membership/src/lib.rs (hunk 1 above)
Comment (Dinonard) [reply substrate/frame/membership/src/lib.rs:503]: I understand that but it's redundant - there's no dependency on `m` related to the weight.
Seems wrong to leave it, especially since it's a minute change.

I will also update westend/rococo/example-node not to break anything.

If you strongly suggest to leave it as is, I'll do it like that.
But in that case I should also revert the rename of `reset_member` to keep it compatible.

Comment (Dinonard) [reply substrate/frame/membership/src/lib.rs:503]: Thanks for the fast review btw 🙂

**Review by bkchr — APPROVED** [review]

File: substrate/frame/membership/src/lib.rs [hunk 3]
```diff
@@ -267,13 +276,19 @@ pub mod pallet {
...
+		pub fn change_key(
+			origin: OriginFor<T>,
+			new: AccountIdLookupOf<T>,
+		) -> DispatchResultWithPostInfo {
 			let remove = ensure_signed(origin)?;
 			let new = T::Lookup::lookup(new)?;
 
+			let mut members_length = T::MaxMembers::get();
+
 			if remove != new {
```
[the remark this answers is not in the document]
Comment (ggwpez) [reply substrate/frame/membership/src/lib.rs:289]: Yea. Maybe even return an error... but dont want to side-track this MR.

[the remark this answers is not in the document]
Comment (bkchr) [reply substrate/frame/membership/src/lib.rs:289]: Yeah would be nice! Ty!

[the remark this answers is not in the document]
Comment (Dinonard) [reply substrate/frame/membership/src/lib.rs]: Done!

I opted for the same approach as in swap_member to keep it consistent, and to minimize changes.

**Review by ggwpez — APPROVED** [review]

**Comment by Dinonard** [comment]
Can this be merged or is there something else I should do?
